astropy/io/fits/tests/test_image.py

```diff
@@ -926,7 +926,7 @@ def test_image_update_header(self):
             assert (orig_data == hdul[1].data).all()
 
     # The test below raised a `ResourceWarning: unclosed transport` exception
-    # due to a bug in Python <=3.10 (cf. cpython#90476)
+    # due to a bug in PYTHON_LT_3_11 (cf. cpython#90476)
```

Comment:
The intention behind having these `PYTHON_LT_*` variables was that when we drop support for some Python version then we can simply delete the definition of the corresponding variable causing all the code that is using it to raise an error, and this would prevent compatibility code from staying around longer than it is needed. But this setup also means that strings like that should never be grepped for, so I'm not convinced this change is helpful.

I also think we should stop using this `PYTHON_LT_*` variables because we are now using Ruff and its rule [UP036](https://docs.astral.sh/ruff/rules/outdated-version-block/) can recognize comparisons involving `sys.version_info` calls and update the code automatically (possibly requiring the `--unsafe-fixes` option).

The best thing to do here would be to replace the comment with code that explicitly uses `sys.version_info`.

Reply:
For the numpy equivalent, I tend to do the `find astropy -name \*.py -exec grep NUMPY_LT...` - but if it is easy to refactor with a version check, that would indeed be ideal.

Reply:
Amusingly, I wrote UP036 in the original (pyupgrade) code, so I'm well aware it exists 😅 
However my goal here is absolutely not to revolutionise how astropy code is written, and I believe it would be best left out of this PR ?

Reply:
Thanks, learn something new every day!

Reply:
Initially I misread the comment and *thought* it was not relevant in 3.10.  Happy to just leave it out of this PR.

> But overall it would be good to write (or rewrite) any compatibility code in a way that prevents it from outliving its usefulness.

for the record I agree (this is what I wrote UP036 for), but I think I recall there was some objections when I proposed this refactor in astropy some time ago (but I can't remember the exact context).

Reply:
(I dropped the change)

Reply:
FWIW, what I remember about auto-upgrade is mostly that some of us (including me) felt that one should not upgrade code just for the sake of it (i.e., one doesn't *have to* use `|=` to update a dict, using `.update()` is fine).

Automatically recognizing pieces of code that are no longer useful does not fall in that category, though!

Reply:
I'll open another PR to propose this.

Reply:
for posterity: this is #16186